src/connection_manager.c

```diff
@@ -859,16 +861,23 @@ qd_error_t qd_entity_refresh_connector(qd_entity_t* entity, void *impl)
     case CXTR_STATE_INIT:
       state_info = "INITIALIZING";
       break;
+    case CXTR_STATE_DELETED:
+      state_info = "DELETED";
+      break;
     default:
       state_info = "UNKNOWN";
       break;
     }
 
     if (qd_entity_set_string(entity, "failoverUrls", failover_info) == 0
-            && qd_entity_set_string(entity, "connectionStatus", state_info) == 0
-            && qd_entity_set_string(entity, "connectionMsg", ct->conn_msg) == 0)
+        && qd_entity_set_string(entity, "connectionStatus", state_info) == 0
+        && qd_entity_set_string(entity, "connectionMsg", ct->conn_msg) == 0) {
+
+        sys_mutex_unlock(ct->lock);
```

Comment:
can we change this to connector->lock ? In some places we have ct->lock and in some places we have connector->lock. The connector->lock is easier to understand IMO.